project/Settings.scala

```diff
@@ -24,32 +24,29 @@ import scalariform.formatter.preferences._
 
 object Settings {
 
-  val JavaVersion = "1.6"
-  val SVersion = "2.11.8"
+  val JavaVersion = "1.8"
+  val SVersion = "2.12.1"
 
   lazy val basicSettings = Seq(
     ivyScala := ivyScala.value map { _.copy(overrideScalaVersion = true) },
-    scalaVersion                    := SVersion,
-    crossScalaVersions              := Seq("2.10.5", SVersion),
-    resolvers                       ++= Dependencies.resolutionRepos,
-    fork in run                     := true,
-    parallelExecution in Global     := false,
-    testGrouping in Test            := singleTestPerJvm((definedTests in Test).value, (javaOptions in Test).value),
-    javacOptions                    := Seq(
+    scalaVersion                        := SVersion,
+    crossScalaVersions                  := Seq("2.10.5", "2.11.8", "2.12.1"),
+    resolvers                           ++= Dependencies.resolutionRepos,
+    fork in run                         := true,
+    parallelExecution in Global         := false,
+    testGrouping in Test                := singleTestPerJvm((definedTests in Test).value, (javaOptions in Test).value),
+    javacOptions                        := Seq(
       "-Xlint:-options",
       "-source", JavaVersion, "-target", JavaVersion),
```

Comment:
I believe you are correct. I misconstrued the warnings from scalac; javac options can remain as 1.6.